### PR TITLE
👷 setup Solidity code formatter

### PR DESCRIPTION
**Pull request by wenceslas-sanchez**
@poyea
Just to let you @mkubdev [prettier](https://github.com/prettier-solidity/prettier-plugin-solidity) was used because the repo is more active. (/= [Solhint](https://github.com/protofire/solhint)) 😄

### CODE REVIEW

File: .github/workflows/autoprettier.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        id: git-check
+        run: |
+          git config --global user.name 'autoprettier'
+          git config --global user.email 'actions@github.com'
+          echo ::set-output name=modified::$(if git diff-index --quiet HEAD --; then echo "false"; else echo "true"; fi)
+      - name: Commit if any changes.
+        if: steps.git-check.outputs.modified == 'true'
+        run: |
+          git commit -am "🎨 Format Solidity code with Prettier"
+          git push
```
Comment (Panquesito7) [inline .github/workflows/autoprettier.yml:45]: Missing an endline.
```suggestion
          git push

```

**Review by mkubdev — APPROVED** [review]
So cool! :fire:

**Review by Panquesito7 — APPROVED** [review]
Looks good, thanks! 🚀 A question: is this ready or still a WIP?

**Comment by wenceslas-sanchez** [comment]
> Looks good, thanks! 🚀 A question: is this ready or still a WIP?

I tried it locally and it works ! You can merge it I guess 😄

**Comment by wenceslas-sanchez** [comment]
The action failed, and I faced the same issue with another project. Here is how I resolve it (and make the litter works) : 
- delete .yml file
- run locally `npx prettier --write .` then commit/push
- finally re-add the .yml file

https://www.koskila.net/how-to-resolve-refusing-to-allow-an-integration-to-create-or-update-github-workflows-main-yml-on-github-desktop/

**Pull request merged**